### PR TITLE
Feature: implement credential status check

### PR DESCRIPTION
**Pull request by mirceanis**
### What's here
This adds the necessary code and some tests to check a credential against an on-chain registry.

* multinetwork by default
* custom configuration similar to ethr-did-resolver, except default provider is localhost!
* easy configuration using [`infuraProjectId`](https://infura.io/dashboard)

### Usage:
Normally, this would be used in conjunction with the [`credential-status`](https://github.com/uport-project/credential-status) library, so besides configuring it there's not much else to do:

```typescript
import { EthrStatusRegistry } from 'ethr-status-registry'
import { Status } from 'credential-status'

const status = new Status({
    ...new EthrStatusRegistry({infuraProjectId: 'YOUR Infura PROJECT ID HERE'}).asStatusMethod,
})

const result = await status.checkStatus(token)
// { "revoked": true }
```

But it can also be used independently:

```typescript
const statusReg = new EthrStatusRegistry(config)
const revocationStatus = statusReg.checkStatus(token)
// {revoked : false}
```

The [readme](https://github.com/uport-project/ethr-status-registry/blob/9587f9ca4dc7a982f7bf9e92ef6c7938bdb459bb/README.md) shows more detailed examples

### CODE REVIEW

File: src/contracts/ethr-status-registry.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    "outputs": [
+      {
+        "name": "",
+        "type": "bool"
+      }
+    ],
+    "payable": false,
+    "stateMutability": "view",
+    "type": "function"
+  }
```
Comment (ajunge) [inline src/contracts/ethr-status-registry.json]: You only need this part of the ABI. The rest can be safely be deleted

File: src/index.ts [hunk 2]
```diff
@@ -42,7 +40,7 @@ export class EthrStatusRegistry implements StatusResolver {
 
   checkStatus(credential: string): Promise<null | CredentialStatus> {
     const decodedJWT = decodeJWT(credential).payload as JWTDecodedExtended
-    if (decodedJWT.status && decodedJWT.status.type === this.methodName) {
+    if (decodedJWT.status?.type === this.methodName) {
```
Comment (mirceanis) [inline src/index.ts]: I was hoping this operator would become part of the language!
Thanks

File: src/contracts/ethr-status-registry.json (hunk 1 above)
Comment (mirceanis) [reply src/contracts/ethr-status-registry.json]: Good catch, thank you

**Comment by mirceanis** [comment]
> Haven't run the code but just looking at it looks good.
> Question: The result is only revoked/not revoked? It can be checked who revoked it?

@ajunge right now only issuer revocations are considered by this lib, mostly because of phrasing in W3C spec.
So, this implementation only answers the question "has this issuer revoked this credential?"

But the contract supports any kind of revocation. I made some notes in the [README](https://github.com/uport-project/ethr-status-registry/blob/9587f9ca4dc7a982f7bf9e92ef6c7938bdb459bb/README.md#potential-improvements) with potential improvements

**Pull request merged**